pkgs/by-name/pm/pmars/package.nix

```
@@ -0,0 +1,110 @@
+{
+  stdenv,
+  lib,
+  fetchzip,
+  installShellFiles,
+  libX11,
+  ncurses,
+  pkg-config,
+  graphxMode ? "${if stdenv.hostPlatform.isLinux then "XWIN" else "CURSES"}GRAPHX",
+}:
+
+let
+  options = [
+    graphxMode
+    "EXT94"
+    "PERMUTATE"
+    "RWLIMIT"
+  ];
+  pkgConfigLibs =
+    lib.optionals (graphxMode == "XWINGRAPHX") [ "x11" ]
+    ++ lib.optionals (
+      graphxMode == "CURSESGRAPHX" || (!stdenv.hostPlatform.isLinux && graphxMode == "GRAPHX")
+    ) [ "ncurses" ];
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "pmars";
+  version = "0.9.4";
+
+  src = fetchzip {
+    url = "http://www.koth.org/pmars/pmars-${finalAttrs.version}.zip";
+    hash = "sha256-68zsH9HWWp13pozjMajayS/VhY8iTosUp1CvcAmj/dE=";
+  };
+
+  patches = [
+    # redefinition of 'round' as different kind of symbol
```

documentation nit: does this prevent any error? right now it wasn't super clear to me why this patch was here

---

Yes, the error happens specifically under Clang because it's more strict about the re-use of the generic `round` name than GCC:

```
Compiling disasm.o
In file included from disasm.c:26:
sim.h:92:12: warning: built-in function 'round' declared as non-function [-Wbuiltin-declaration-mismatch]
   92 | extern int round;
      |            ^~~~~
```

vs

```
Compiling disasm.o
[warnings]
In file included from disasm.c:26:
./sim.h:92:12: error: redefinition of 'round' as different kind of symbol
   92 | extern int round;
      |            ^
```

How about:

```suggestion
    # Error under Clang due to global "round" variable: redefinition of 'round' as different kind of symbol
```

?

---

that's pretty clear - thanks!